src/modules/commander/commander.cpp

```diff
@@ -2446,56 +2447,32 @@ Commander::run()
-			 * and both failed we want to terminate the flight */
-			if (internal_state.main_state != commander_state_s::MAIN_STATE_MANUAL &&
-			    internal_state.main_state != commander_state_s::MAIN_STATE_ACRO &&
-			    internal_state.main_state != commander_state_s::MAIN_STATE_RATTITUDE &&
-			    internal_state.main_state != commander_state_s::MAIN_STATE_STAB &&
-			    internal_state.main_state != commander_state_s::MAIN_STATE_ALTCTL &&
-			    internal_state.main_state != commander_state_s::MAIN_STATE_POSCTL &&
-			    status.data_link_lost) {
-
-				armed.force_failsafe = true;
-				status_changed = true;
-				static bool flight_termination_printed = false;
+		/* Check for failure detector status */
+		if (armed.armed) {
 
-				if (!flight_termination_printed) {
-					mavlink_log_critical(&mavlink_log_pub, "DL and GPS lost: flight termination");
-					flight_termination_printed = true;
-				}
+			if (_failure_detector.update()) {
 
-				if (counter % (1000000 / COMMANDER_MONITORING_INTERVAL) == 0) {
-					mavlink_log_critical(&mavlink_log_pub, "DL and GPS lost: flight termination");
-				}
-			}
+				const auto _failure_status = _failure_detector.get();
 
-			/* At this point the rc signal and the gps system have been checked
-			 * If we are in manual (controlled with RC):
-			 * if both failed we want to terminate the flight */
-			if ((internal_state.main_state == commander_state_s::MAIN_STATE_ACRO ||
-			     internal_state.main_state == commander_state_s::MAIN_STATE_RATTITUDE ||
-			     internal_state.main_state == commander_state_s::MAIN_STATE_MANUAL ||
-			     internal_state.main_state == commander_state_s::MAIN_STATE_STAB ||
-			     internal_state.main_state == commander_state_s::MAIN_STATE_ALTCTL ||
-			     internal_state.main_state == commander_state_s::MAIN_STATE_POSCTL) &&
-			    status.rc_signal_lost) {
-
-				armed.force_failsafe = true;
-				status_changed = true;
-				static bool flight_termination_printed = false;
+				if (_failure_status.roll ||
+					_failure_status.pitch) {
 
-				if (!flight_termination_printed) {
-					warnx("Flight termination because of RC signal loss and GPS failure");
-					flight_termination_printed = true;
-				}
+					armed.force_failsafe = true;
+					status_changed = true;
+
+					// Only display an user message if the circuit-breaker is disabled
+					if (!status_flags.circuit_breaker_flight_termination_disabled) {
+						static bool parachute_termination_printed = false;
+
+						if (!parachute_termination_printed) {
+							warnx("Flight termination because of attitude exceeding maximum values");
+							parachute_termination_printed = true;
+						}
 
-				if (counter % (1000000 / COMMANDER_MONITORING_INTERVAL) == 0) {
-					mavlink_log_critical(&mavlink_log_pub, "RC and GPS lost: flight termination");
+						if (counter % (1000000 / COMMANDER_MONITORING_INTERVAL) == 0) {
+							mavlink_log_critical(&mavlink_log_pub, "Attitude failure detected: force failsafe");
```

Comment:
Printing continuously? As long as the HEARTBEAT system_status is updated correctly to MAV_STATE_FLIGHT_TERMINATION I don't think we should send this continuously.

Reply:
Do I move that `mavlink_log_critical` in the `if (!parachute_termination_printed)` above and remove the `warnx`/`PX4_WARN` ?